### PR TITLE
<amp-pinterest> Sanitize URLs

### PR DESCRIPTION
**Pull request by nainar**
Currently we call openWindowDialog on a click on an AMP widget. These `href`s should be sanitized before being passed to openWindowDialog.

### CODE REVIEW

File: extensions/amp-pinterest/0.1/pin-widget.js [hunk 1]
```diff
@@ -58,9 +58,9 @@ export class PinWidget {
     const log = el.getAttribute('data-pin-log');
     if (href) {
       if (shouldPop) {
-        openWindowDialog(window, href, '_pinit', POP);
+        openWindowDialog(window, encodeURI(href), '_pinit', POP);
```
Comment (jridgewell) [inline extensions/amp-pinterest/0.1/pin-widget.js]: This won't do it. I can still use a `javascript:alert('PWNED')`, which'll pass right through. We likely want something like `assertAbsoluteHttpOrHttpsUrl`.

File: extensions/amp-pinterest/0.1/pin-widget.js [hunk 2]
```diff
@@ -57,6 +57,7 @@ export class PinWidget {
     const href = el.getAttribute('data-pin-href');
     const log = el.getAttribute('data-pin-log');
     if (href) {
+      assertAbsoluteHttpOrHttpsUrl(href);
```
Comment (nainar) [inline extensions/amp-pinterest/0.1/pin-widget.js]: Asserting that the href is starting with http/https here. 

Won't allow localhost - so testing 🤷
Also can one still force code in the latter part of the URL?

File: extensions/amp-pinterest/0.1/pin-widget.js (hunk 1 above)
Comment (nainar) [reply extensions/amp-pinterest/0.1/pin-widget.js]: Taken this out in favor of assertAbsoluteHttpOrHttpsUrl

**Comment by nainar** [comment]
Can you please merge it in too? I don't have the required rights.

**Comment by jridgewell** [comment]
Ping me when travis passes.

**Comment by nainar** [comment]
🔔🔔🔔

**Pull request merged**